crates/utils/src/utils/validation.rs

```diff
@@ -152,7 +152,8 @@ pub fn is_valid_matrix_id(matrix_id: &str) -> LemmyResult<()> {
 
 pub fn is_valid_post_title(title: &str) -> LemmyResult<()> {
   let length = title.trim().chars().count();
-  let check = (3..=200).contains(&length) && !has_newline(title);
+  let check =
+    (3..=200).contains(&length) && !has_newline(title) && has_3_permitted_display_chars(title);
```

Comment:
Curiously, `has_3_premitted_display_chars` still allows users to have invisible characters, but enforces that there needs to be at least 3 visible characters. It's currently used for validating display names.

Reply:
The trans flag is two unicode emojies (4 chars) in a u200d trench coat

Reply:
See https://github.com/LemmyNet/lemmy/issues/3888 for context about has_3_premitted_display_chars and also see the docs for [str.chars()](https://doc.rust-lang.org/std/primitive.str.html#method.chars). It seems we would need something like [unicode-segmentation](https://crates.io/crates/unicode-segmentation) to iterate over grapheme clusters and filter invisible ones, so that combined symbols like flags are not affected. Ideally there would be a library to check for this, but the only one I found ([invisible_unicode](https://crates.io/crates/invisible_unicode)) seems to have the same problem and is not actively maintained.

Reply:
Here's the json list of characters if someone wants to do this: https://github.com/flopp/invisible-characters/blob/main/characters.json .

Reply:
@SleeplessOne1917 I've created this now: https://crates.io/crates/invisible-characters

Reply:
Make sure to add a test case with some flags and emojis, I suspect that it will fail with this simple .contains() check.